### ISSUE TITLE
Update main MoBIE update site to match MoBIE-beta

### ISSUE THREAD

**Issue by K-Meech** [mobie/mobie-viewer-fiji#497]
We will be updating the main MoBIE update site to match MoBIE-beta in the next few days. So be aware that some things may stop working with the main MoBIE update site, as the relevant repositories etc are updated!

In the meantime, you can still use MoBIE-beta, and we'll let you know when everything is moved over :)

@constantinpape, @tischi - feel free to tag anyone else I've missed!
@martinschorb @vzinche @KateMoreva @yanschwab

**Comment by tischi** [comment]
@K-Meech in the mobie-io repo the `master` branch is now called `main`. should we, based on the current `develop` branch, also make a new branch `main` here and deprecate the old `master` branch?

**Comment by constantinpape** [comment]
> should we, based on the current `develop` branch, also make a new branch `main` here and deprecate the old `master` branch?

It's better to just rename the branch from `master` to `main`: https://www.cds.caltech.edu/~mhucka/blog/posts/2020-07-19/renaming-your-master-branch-in-git/

In addition, we will need to update the CI:
https://github.com/mobie/mobie-viewer-fiji/blob/master/.github/workflows/build-main.yml#L6
https://github.com/mobie/mobie-viewer-fiji/blob/master/.github/workflows/build-pr.yml#L6

That's all the changes necessary as far as I can see; not sure if there's something that has to change for the FIJI update side.

**Comment by tischi** [comment]
Done. I did not yet dare to delete the old master branch (in fact, we don't have to, but it would be nice to avoid confusion). This can be done like this: `git push origin :master`

**Comment by constantinpape** [comment]
> Done. I did not yet dare to delete the old master branch (in fact, we don't have to, but it would be nice to avoid confusion). This can be done like this: `git push origin :master`

Please delete it as soon as possible, otherwise this will be maximally confusing.

**Comment by tischi** [comment]
@constantinpape 

I would now be ready to change the MoBIE update site.

- [X] merged `develop` into `main`
- [X] deleted `master`

Should I update the MoBIE update site now?

**Comment by constantinpape** [comment]
@tischi yes, please go ahead!

**Comment by tischi** [comment]
I will document the process here, not to loose track. I will explictely ping someone if I need help.

**Comment by tischi** [comment]
These are the jars that we currently upload to **MoBIE-beta**
<img width="892" alt="image" src="https://user-images.githubusercontent.com/2157566/141468513-994d3a6e-d3da-404e-b76d-b02b178c1d33.png">

**Comment by tischi** [comment]
These are the jars that we are used to upload to **MoBIE**

<img width="892" alt="image" src="https://user-images.githubusercontent.com/2157566/141468675-0610bb8c-6ada-4594-846f-700038b1fc55.png">

**Comment by tischi** [comment]
- [ ] remove mobie.jar from MoBIE
- [ ] add the additional jars to MoBIE

**Comment by tischi** [comment]
- [ ] do we really need to upload bigvolumeviewer?
- [ ] do we really need to upload spim_data?
- [ ] somehow bigdataviewer-spimdata-extras was missing when trying to upload to MoBIE...

**Comment by tischi** [comment]
<img width="892" alt="image" src="https://user-images.githubusercontent.com/2157566/141470508-fde20024-bc13-4ff6-903a-9b4bfd8263dc.png">

@constantinpape 

Done. Could you please test?

- [ ] `Plugins › MoBIE › Open › Open PlatyBrowser` is not working

**Comment by constantinpape** [comment]
>     * `Plugins › MoBIE › Open › Open PlatyBrowser` is not working

Yes, I still need to update the git repo for this, I will check now and then report back.

**Comment by tischi** [comment]
@K-Meech @KateMoreva 

Please note that for the rest of the day I will fix bugs directly and only on `MoBIE/main`.
I will then at the end of the day synchronize `MoBIE-beta/develop`.

**Comment by constantinpape** [comment]
@tischi I went through all projects now and updated the ones where `main` was not on the new spec yet.
Everything can be opened! (Except https://github.com/mobie/plankton-fibsem-project/issues/4, but I think that is an issue with the project, I will investigate later).

**Comment by tischi** [comment]
> Everything can be opened! 

Also the PlatyBrowser?

**Comment by constantinpape** [comment]
> Also the PlatyBrowser?

Yes

**Comment by tischi** [comment]
a w e s o m e ! ! !

good end of the week

good start in the weekend

By the way: another _very good news_ is that we are now [running tests with github actions](https://github.com/mobie/mobie-viewer-fiji/issues/520) that involve opening the whole MoBIE UI.

**Comment by tischi** [comment]
Special thanks to @KateMoreva for all the great work and bug fixes in `mobie-io` 👏

**Comment by KateMoreva** [comment]
> Special thanks to @KateMoreva for all the great work and bug fixes in `mobie-io` clap

Thank you @tischi, @K-Meech, @constantinpape and @martinschorb, @vzinche, @yanschwab for the awesome work!

**Comment by constantinpape** [comment]
> By the way: another _very good news_ is that we are now [running tests with github actions](https://github.com/mobie/mobie-viewer-fiji/issues/520) that involve opening the whole MoBIE UI.

That's awesome. Also seems like a good opportunity to fix all the tests now ;): https://github.com/mobie/mobie-viewer-fiji/actions/runs/1453039795

**Comment by tischi** [comment]
> Also seems like a good opportunity to fix all the tests now ;)

The tests were in fact running for this build. The "error" (which is not really an error, imho) is due to the fact that this specific version of MoBIE is already deployed to scijava-maven and then it throws:

```
Error:  Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:2.8.2:deploy (default-deploy) on project mobie-viewer-fiji: Failed to deploy artifacts: Could not transfer artifact org.embl.mobie:mobie-viewer-fiji:jar:2.0.6 from/to scijava.releases (dav:https://maven.scijava.org/content/repositories/releases): Failed to transfer file: https://maven.scijava.org/content/repositories/releases/org/embl/mobie/mobie-viewer-fiji/2.0.6/mobie-viewer-fiji-2.0.6.jar. Return code is: 400 -> [Help 1]
```

It is a bit annoying that one needs to scroll all the way through the build step to see why there is an error.

TLDR: Whenever we push to `develop` or `main` and want the github actions to fully complete we need to bump the version (cc @K-Meech @KateMoreva).

...maybe this makes sense anyway, because changing `develop` and `main` should only be done if we have some production ready changes. Everything else should happen in other branches.

**Comment by constantinpape** [comment]
> TLDR: Whenever we push to `develop` or `main` and want the github actions to fully complete we need to bump the version (cc @K-Meech @KateMoreva).
> 
> ...maybe this makes sense anyway, because changing `develop` and `main` should only be done if we have some production ready changes. Everything else should happen in other branches.

I think it would be better to only deploy to the update sites if we actually make a new release. That should be relatively easy by creating a separate action for deploying to the update site(s) and adding the trigger `on: releases` (probably needs some more logic to distinguish whether it should be deployed to the main or beta update site, e.g. by checking on which branch the release is).

**Comment by tischi** [comment]
> I think it would be better to only deploy to the update sites if we actually make a new release

Deploying to scijava-maven is different from deploying to the update site. Deploying to the update site has to be done manually.

**Comment by constantinpape** [comment]
> > 
> 
> Deploying to scijava-maven is different from deploying to the update site. Deploying to the update site has to be done manually.

I see. But this doesn't matter re my comment above, just replace deploy to update site with deploy to scijava-maven.

**Comment by tischi** [comment]
@constantinpape @K-Meech @KateMoreva 

To clarify this again: when I say "deploying to scijava-maven" I mean that one can depend on the jar via pom files, because it is [served here](https://maven.scijava.org/#nexus-search;quick~mobie-viewer-fiji). 

This has (unf.) nothing to do with what [we ship on the Fiji update site](https://sites.imagej.net/MoBIE/)

**Comment by tischi** [comment]
> That should be relatively easy by creating a separate action for deploying to the update site(s) and adding the trigger on: releases

That sounds interesting...  THB I don't know how to make a release from the git command line. Is this possible or does one need to do this on the GitHub website?

**Comment by constantinpape** [comment]
> THB I don't know how to make a release from the git command line.

That's done with `git tag` (+ some arguments ofc). But you don't necessarily need to do this via the command line, it's fine to do it via the website.

Usually the workflow is the following:
- developer makes a release (either via github website or git tag)
- CI is triggered by the release (using the `on: releases` trigger for GH actions)
- CI does everything required for a new release, here deploying to scijava.maven

(There are ofc other options, like using the version defined in some file to trigger the release, in python that's `__version__`, not sure what the corresponding thing in Java would be, and automatically drafting releases and triggering deployment when this version is changed by a commit. And there are tools like [bumpversion](https://github.com/peritus/bumpversion) to help automating this; but I don't think we would need this here.)

**Comment by tischi** [comment]
This makes sense. I change it in both the main and the develop branch: https://github.com/mobie/mobie-viewer-fiji/blob/6f3ffad95eb87b6e93fd23b04cf86c13a47fa7c8/.github/workflows/build-main.yml#L4

**Comment by tischi** [comment]
I tried triggering a release from the CLI but did not manage; could you explain in more detail how to do this?

I manged on the GitHub UI. Does the release name make sense?
https://github.com/mobie/mobie-viewer-fiji/releases

Doing it via the GitHub UI is quite some extra work, but maybe we don't make releases for MoBIE-beta? Then it could be OK...

**Comment by tischi** [comment]
Build on release is working: https://github.com/mobie/mobie-viewer-fiji/actions/runs/1456922401

And deployment works: https://maven.scijava.org/#nexus-search;quick~mobie-viewer-fiji

**Comment by K-Meech** [comment]
Cool - sounds good! I guess in general the version in the pom on MoBIE-beta should always be a -SNAPSHOT version, and on main always not a -SNAPSHOT version. If I understand correctly, -SNAPSHOT versions can be deployed to scijava-maven many times, so you wouldn't see these deploy errors for them.

**Comment by tischi** [comment]
Aha, good point. In that case maybe for the develop branch (MoBIE-beta) we can build `on push`?

**Comment by K-Meech** [comment]
I think so! Might be useful so the latest version is still available on maven (I often re-use bits of mobie in my other projects, and this makes it a lot easier). We'd just have to be careful to always make sure it stays as a -SNAPSHOT!

**Comment by tischi** [comment]
@K-Meech 

I changed it to `on push` for the `develop` branch and I made it a SNAPSHOT version.
